helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -89,7 +89,7 @@ protected void processEvent(ClusterEvent event, ResourcesStateMap resourcesState
           + ". Requires HelixManager|DataCache|RESOURCES|CURRENT_STATE|INTERMEDIATE_STATE");
     }
 
-    Map<String, LiveInstance> liveInstances = cache.getLiveInstances();
+    Map<String, LiveInstance> liveInstances = cache.getLiveInstances(true);
```

Comment:
Could this create inconsistency since LivInstances used by all other places in the pipeline are different from here?

Reply:
Duplicate discussion with the above comment.